runway/blueprints/staticsite/auth_at_edge.py

```diff
@@ -0,0 +1,434 @@
+                        EventType='viewer-request',
+                        LambdaFunctionARN=check_auth_lambda_version.ref()
+                    ),
+                    cloudfront.LambdaFunctionAssociation(
+                        EventType='origin-response',
+                        LambdaFunctionARN=http_headers_lambda_version.ref()
+                    )
+                ],
+                TargetOriginId='protected-bucket',
+                ViewerProtocolPolicy='redirect-to-https'
+            ),
+            'DefaultRootObject': 'index.html',
+            'Logging': self.add_logging_bucket(),
+            'PriceClass': variables['PriceClass'],
+            'Enabled': True,
+            'WebACLId': self.add_web_acl(),
+            'CustomErrorResponses': [
+                cloudfront.CustomErrorResponse(
+                    ErrorCode=403,
+                    ResponseCode=200,
+                    ResponsePagePath="/index.html"
+                ),
+                cloudfront.CustomErrorResponse(
+                    ErrorCode=404,
+                    ResponseCode=200,
+                    ResponsePagePath="/index.html"
+                )
+            ],
+            'ViewerCertificate': self.add_acm_cert()
+        }
+
+    def add_cloudfront_distribution(self,
+                                    allow_cloudfront_access,  # type: s3.BucketPolicy
+                                    cloudfront_distribution_options  # type: Dict[str, Any]
+                                    ):
+        # type: (...) -> cloudfront.Distribution
+        """Add the CloudFront distribution to the template / output the id and domain name.
+
+        Keyword Args:
+            allow_cloudfront_access (dict): Allow bucket access resource
+            cloudfront_distribution_options (dict): The distribution options
+
+        Return:
+            dict: The CloudFront Distribution resource
+
+        """
+        distribution = self.template.add_resource(
+            cloudfront.Distribution(
+                'CFDistribution',
+                DependsOn=allow_cloudfront_access.title,
+                DistributionConfig=cloudfront.DistributionConfig(
+                    **cloudfront_distribution_options
+                )
+            )
+        )
+        self.template.add_output(Output(
+            'CFDistributionId',
+            Description='CloudFront distribution ID',
+            Value=distribution.ref()
+        ))
+        self.template.add_output(
+            Output(
+                'CFDistributionDomainName',
+                Description='CloudFront distribution domain name',
+                Value=distribution.get_att('DomainName')
+            )
+        )
+        return distribution
+
+
+# Helper section to enable easy blueprint -> template generation
+# (just run `python <thisfile>` to output the json)
+if __name__ == "__main__":
+    print(AuthAtEdge('test', Context({"namespace": "test"}), None).to_json())
```

Comment:
if this were run, it would fail because of `self.context.hook_data['aae_lambda_config']`. You would need to add error handing when `hook_data` is used or build out your mock `Context` a bit more to include the necessary hook data.

Reply:
That would be an effort to mock. I think I'm going to just remove it for the time being and will make a ticket to address at a later time.

Reply:
Good catch